staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -329,9 +333,18 @@ func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
 
 // OpenAPISchema fetches the open api schema using a rest client and parses the proto.
 func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
+	data, err := d.restClient.Get().AbsPath("/openapi/v2").SetHeader("Accept", mimePb).Do().Raw()
 	if err != nil {
-		return nil, err
+		if errors.IsNotFound(err) {
```

Comment:
a 1.9 server will return a 403 error for this, not a 404

Reply:
Ah, because the path wasn't in ClusterRoles. Nice catch! I tested with `/openapi/v3` which didn't catch that :(

Updated